### PR TITLE
Refactor: refactor AmountExtractor

### PR DESCRIPTION
**Pull request by andreitsukanov**

### CODE REVIEW

File: src/main/java/com/zavod/utils/AmountExtractor.java [hunk 1]
```diff
@@ -18,39 +18,40 @@ private AmountExtractor() {
...
 
   public static BigDecimal extract(MicrotransactionPayload payload) {
-    String amountString = payload.getAmountString();
+    return extract(payload.getAmountString(), payload.getAmount());
+  }
+
+  private static BigDecimal extract(String amountString, double amountValue) {
+    BigDecimal amount = amountString.length() > 0
+        ? new BigDecimal(amountString)
+        : new BigDecimal(amountValue);
```
Comment (emitskevich) [inline src/main/java/com/zavod/utils/AmountExtractor.java:31]: Did you change `BigDecimal.valueOf()` to `new BigDecimal()` intentionally? Behaviour inside is not the same.

File: src/main/java/com/zavod/utils/AmountExtractor.java [hunk 2]
```diff
@@ -18,39 +18,40 @@ private AmountExtractor() {
...
+
+    BigDecimal scaled = amount.setScale(SCALE, HALF_UP);
     if (amountString.length() > 0) {
-      BigDecimal amount = new BigDecimal(amountString);
-      return scale(amount, true);
+      log(amount, scaled);
     }
```
Comment (emitskevich) [inline src/main/java/com/zavod/utils/AmountExtractor.java:36]: Suggestion:
```
    if (amountString.length() > 0 && amount.compareTo(scaled) != 0) {
      logLoss(amount, scaled);
    }
```

Comment (wardomski) [reply src/main/java/com/zavod/utils/AmountExtractor.java]: Applied

File: src/main/java/com/zavod/utils/AmountExtractor.java (hunk 1 above)
Comment (wardomski) [reply src/main/java/com/zavod/utils/AmountExtractor.java]: Fixed

**Review by emitskevich — CHANGES_REQUESTED** [review]
Can you please provide tests for two public methods?

**Review by emitskevich — APPROVED** [review]

**Pull request merged**